### PR TITLE
some fixes

### PR DESCRIPTION
**Pull request by burgerbutts**
Compilied under Clang++ to test out something and it brought up quite a bit of errors/warnings. Everything aside from the HIDDEN_TILDE should be fixed, and there's also a fix for hedit breaking helpfile formatting.

Remaining problems are the ¢ character being a wide char, not being allowed to be defined as '¢'. It should properly be defined as L'¢', but this causes countless problems since no functions support unicode. I've left all this code alone on the git so it all compiles fine on GCC, just a simple warning about it being incorrectly sized.

### CODE REVIEW

**Comment by burgerbutts** [comment]
The command rdelete was causing a ( the bug() call ) bug in the global fix_exits() function when deleting a room from a prototype area. The bug should be solved since the function now doing the opposite of make_room, with the help of the additional area_data*->fix_exits() call.

**Comment by Arthmoor** [comment]
@burgerbutts I have a question not necessarily related to these fixes. I know it's a rather long time since you submitted them, but did you need to alter the Makefile at all? I'm not able to get the dynamic commands to load when compiled in Clang++ 4.

**Comment by burgerbutts** [comment]
I believe so, sorry. I assume you got it fixed in the recent pull you did on the makefile?

Only difference I think I had in my makefile was "-std=c++1z" instead of "-std=c++0x". I think the -rdynamic was the main culprit though.

File: src/area.cpp [hunk 1]
```diff
@@ -3516,9 +3513,8 @@ CMDF( do_aset )
          ch->print( "Usage: aset <filename> gem <percentage>\r\n" );
          return;
       }
-      value = atoi( argument.c_str(  ) );
-      tarea->tg_gem = value;
-      ch->printf( "Area chance to generate gem set to %hu%%\r\n", value );
+      tarea->tg_gem = stoi( argument );
+      ch->printf( "Area chance to generate gem set to %hu%%\r\n", tarea->tg_ge );
```
Comment (burgerbutts) [inline src/area.cpp]: I think the printf function is using the wrong variable here, should probably be tarea->tg_gem

**Comment by Arthmoor** [comment]
It's actually not fixed. The code will get all the way to the end of the compile and then spits this out:

clang: error: unsupported option '--export-dynamic'

If I then block that option, it will complete, but the MUD will start with all the commands broken because they require the dlsym support and it didn't get linked. Everything I could find on Google says the option should work the way I have it specified now for use in clang++

**Pull request merged**